luigi/contrib/oracle.py

```diff
@@ -0,0 +1,299 @@
+    `password`, `table` and `columns` attributes.
+
+    To customize how to access data from an input task, override the `rows` method
+    with a generator that yields each row as a tuple with fields ordered according to `columns`.
+    """
+
+    def rows(self):
+        """
+        Return/yield tuples or lists corresponding to each row to be inserted.
+        """
+        with self.input().open("r") as fobj:
+            for line in fobj:
+                yield line.strip("\n").split("\t")
+
+    # everything below will rarely have to be overridden
+
+    def output(self):
+        """
+        Returns a OracleTarget representing the inserted dataset.
+
+        Normally you don't override this.
+        """
+        return OracleTarget(
+            host=self.host,
+            port=self.port,
+            database=self.database,
+            dsn=self.dsn,
+            user=self.user,
+            password=self.password,
+            table=self.table,
+            update_id=self.update_id,
+        )
+
+    def copy(self, cursor):
+        query = f"""
+            INSERT INTO {self.table} ({','.join(self.columns)})
+            VALUES ({','.join([':'+c for c in self.columns])})
+        """
+        rows = []
+        for idx, row in enumerate(self.rows()):
+            rows.append(row)
+
+            if (idx + 1) % self.bulk_size == 0:
+                cursor.executemany(query, rows)
+                rows = []
+
+        cursor.executemany(query, rows)
+
+    def run(self):
+        """
+        Inserts data generated by rows() into target table.
+
+        If the target table doesn't exist, self.create_table will be called to attempt to create the table.
+
+        Normally you don't want to override this.
+        """
+        if not (self.table and self.columns):
+            raise Exception("table and columns need to be specified")
+
+        connection = self.output().connect()
+
+        # attempt to copy the data into oracle
+        # if it fails because the target table doesn't exist
+        # try to create it by running self.create_table
+        for attempt in range(2):
+            try:
+                cursor = connection.cursor()
+                print("caling init copy...")
+                self.init_copy(connection)
+                self.copy(cursor)
+                self.post_copy(connection)
+                if self.enable_metadata_columns:
+                    self.post_copy_metacolumns(cursor)
+            except oracle.DatabaseError as e:
+                if attempt == 0 and e.args[0].code == "ORA-00942":
+                    self.create_table(connection)
+                else:
+                    raise
+            else:
+                break
+
+        # mark as complete in same transaction
+        self.output().touch(connection)
+        connection.commit()
+        connection.close()
+
+    @property
+    def bulk_size(self):
+        return 10000
```

Comment:
Should this default be configurable from the `[oracle]` cfg section? I'm fine if you think it's best to leave here and override by inheriting this class.

Reply:
actually took it from `mysql` connector its like this in that connector. changed to 
```py
@property
    def bulk_size(self):
        return int(luigi.configuration.get_config().get("oracle", "bulk-size", 10000))
```